src/lua/api-gateway/validation/oauth2/userProfileValidator.lua

```diff
@@ -192,10 +194,12 @@ function _M:validateUserProfile()
             cachedUserProfile = cjson.decode(cachedUserProfile)
         end
         self:setContextProperties(self:getContextPropertiesObject(cachedUserProfile))
-        if ( self:isProfileValid(cachedUserProfile) == true ) then
+
+        local isValid, failureErrorCode, failureMessage = self:isProfileValid(cachedUserProfile)
```

Comment:
It looks as though this will return nil, nil instead of failure code and message.

Reply:
Please note that isProfileValid will be overridden in a "subclass".